app/models/concerns/mergeable.rb

```diff
@@ -24,20 +24,28 @@ def compute_merge_status(attributes)
     end
 
     def merge(attributes)
-      new_record = new(attributes)
-      existing_record = with_discarded.find_by(id: attributes["id"])
+      retries = 0
+
+      begin
+        new_record = new(attributes)
+        existing_record = with_discarded.find_by(id: attributes["id"])
+
+        case compute_merge_status(attributes)
+        when :discarded
+          discarded_record(existing_record)
+        when :invalid
+          invalid_record(new_record)
+        when :new
+          create_new_record(attributes)
+        when :updated
+          update_existing_record(existing_record, attributes)
+        when :old
+          return_old_record(existing_record)
+        end
+      rescue ActiveRecord::RecordNotUnique
+        retries += 1
 
-      case compute_merge_status(attributes)
-      when :discarded
-        discarded_record(existing_record)
-      when :invalid
-        invalid_record(new_record)
-      when :new
-        create_new_record(attributes)
-      when :updated
-        update_existing_record(existing_record, attributes)
-      when :old
-        return_old_record(existing_record)
+        retries <= 1 ? retry : raise
```

Comment:
Should we also raise the original exception back instead of blank top-level one?

Reply:
I believe `raise` with no args reraises the same exception. I will confirm.